### PR TITLE
ncspot: add module

### PR DESCRIPTION
**Pull request by marsam**
### Description

[ncspot](https://github.com/hrkfdn/ncspot) is a ncurses Spotify client written in Rust using librespot.

### Checklist

- [x] Change is backwards compatible.

- [x] Code formatted with `./format`.

- [x] Code tested through `nix-shell --pure tests -A run.all`.

- [ ] Test cases updated/added. See [example](https://github.com/nix-community/home-manager/commit/f3fbb50b68df20da47f9b0def5607857fcc0d021#diff-b61a6d542f9036550ba9c401c80f00ef).

- [x] Commit messages are formatted like

    ```
    {component}: {description}

    {long description}
    ```

    See [CONTRIBUTING](https://github.com/nix-community/home-manager/blob/master/doc/contributing.adoc#sec-commit-style) for more information and [recent commit messages](https://github.com/nix-community/home-manager/commits/master) for examples.

- If this PR adds a new module

  - [x] Added myself as module maintainer. See [example](https://github.com/nix-community/home-manager/blob/068ff76a10e95820f886ac46957edcff4e44621d/modules/programs/lesspipe.nix#L6).

  - [x] Added myself and the module files to `.github/CODEOWNERS`.

### CODE REVIEW

File: modules/programs/ncspot.nix [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+      description = "The package to use for ncspot.";
+    };
+
+    settings = mkOption {
+      type = tomlFormat.type;
+      default = { };
+      example = {
+        shuffle = true;
+        gapless = true;
+      };
```
Comment (berbiche) [inline modules/programs/ncspot.nix:30]:
```suggestion
      example = literalExample ''{
        shuffle = true;
        gapless = true;
      }'';
```

**Review by berbiche — APPROVED** [review]
Thanks for the contribution

File: modules/programs/ncspot.nix (hunk 1 above)
Comment (marsam) [reply modules/programs/ncspot.nix]: Fixed, thanks!

**Comment by sumnerevans** [comment]
@marsam now that #1943 is merged, you should be able to rebase on `master` and the build should pass.

**Comment by berbiche** [comment]
Can you add a news entry to `modules/misc/news.nix`?

**Comment by marsam** [comment]
Sure, added an entry

**Comment by sumnerevans** [comment]
@marsam looks like there's a merge conflict. Can you please rebase on the latest master?

**Comment by marsam** [comment]
Sure! rebased :)

**Review by berbiche — APPROVED** [review]
Thanks!

**Pull request merged**